### ISSUE TITLE
TypeError - no implicit conversion of Array into String

### ISSUE THREAD

**Issue by LucaKIN** [CocoaPods/CocoaPods#7441]
I have some problems trying to add cocoa pods to my project i used that before and worked well but now im having this error.  I tried removing cocoapods, and all linked thing related with that, installing again, creating default cocoapod file, and the error still persist.
I ran pod install with verbose mode but isnt different error message.

### Verbose mode

Analyzing dependencies

Inspecting targets to integrate
  Using `ARCHS` setting to build architectures of target `Pods-ProjectMobile`: (`$(ARCHS_STANDARD)`)
  Using `ARCHS` setting to build architectures of target `Pods-ProjectMobileTests`: (``)

Resolving dependencies of `Podfile`

Comparing resolved specification to the sandbox manifest

Downloading dependencies
  - Running pre install hooks

Generating Pods project
  - Creating Pods project
  - Adding source files to Pods project
  - Adding frameworks to Pods project
  - Adding libraries to Pods project
  - Adding resources to Pods project
  - Linking headers
  - Installing targets
    - Installing target `Pods-ProjectMobile` iOS 8.0

### Command

```
/usr/local/bin/pod install --verbose
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 1.4.0
        Ruby : ruby 2.5.0p0 (2017-12-25 revision 61468) [x86_64-darwin17]
    RubyGems : 2.7.6
        Host : Mac OS X 10.13.3 (17D47)
       Xcode : 9.2 (9C40b)
         Git : git version 2.14.3 (Apple Git-98)
Ruby lib dir : /usr/local/Cellar/ruby/2.5.0_1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ e59338b9a87376da0ae91ad7b742e5da5f578170
```

### Plugins

```
cocoapods-clean       : 0.0.1
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'ProjectMobile' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for ProjectMobile

  target 'ProjectMobileTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

### Error

```
TypeError - no implicit conversion of Array into String
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:118:in `gsub'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:118:in `expand_build_setting'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:97:in `block in resolve_build_setting'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:96:in `each'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:96:in `reduce'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:96:in `resolve_build_setting'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:60:in `debug?'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/object/build_configuration.rb:68:in `type'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/project_helper.rb:227:in `block in configuration_list'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/project_helper.rb:222:in `each'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/project_helper.rb:222:in `configuration_list'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project/project_helper.rb:51:in `new_target'
/usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6/lib/xcodeproj/project.rb:686:in `new_target'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:48:in `add_target'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:15:in `block in install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/user_interface.rb:142:in `message'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:14:in `install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator.rb:175:in `block (2 levels) in install_libraries'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator.rb:173:in `each'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator.rb:173:in `block in install_libraries'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/user_interface.rb:142:in `message'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator.rb:167:in `install_libraries'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer/xcode/pods_project_generator.rb:65:in `generate!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer.rb:184:in `block in generate_pods_project'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/user_interface.rb:64:in `section'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer.rb:183:in `generate_pods_project'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/installer.rb:119:in `install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/command/install.rb:41:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/lib/cocoapods/command.rb:52:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.4.0/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by dnkoutso** [comment]
Is this is a dup of https://github.com/CocoaPods/CocoaPods/issues/7421? Can you please try master version of the https://github.com/CocoaPods/Xcodeproj gem?

**Comment by LucaKIN** [comment]
@dnkoutso same issue than 7421, thk! i had to manually clone cocoapod/xcodeproj and replace xcodeproj-1.5.6 by the new one in /usr/local/lib/ruby/gems/2.5.0/gems/xcodeproj-1.5.6

its closed!

**Comment by dnkoutso** [comment]
I think I understood correctly you mean this is fixed by applying the #7421 change.
